internal/mlflow_test.go

```diff
@@ -162,6 +137,35 @@ func TestAccRegistryWebhooks(t *testing.T) {
 	assert.True(t, len(all) >= 1)
 }
 
+func deleteModelVersion(t *testing.T, w *databricks.WorkspaceClient, ctx context.Context, created *ml.CreateModelVersionResponse) {
+	// Only delete if model version doesn't have "PENDING_REGISTRATION" status else it could lead to test failure due to race condition
+	startTime := time.Now()
+	for {
+		currentModelVersion, err := w.ModelRegistry.GetModelVersion(ctx, ml.GetModelVersionRequest{
+			Name:    created.ModelVersion.Name,
+			Version: created.ModelVersion.Version,
+		})
+		require.NoError(t, err)
+
+		if currentModelVersion.ModelVersion.Status != ml.ModelVersionStatusPendingRegistration {
+			break
+		}
+		currentTime := time.Now()
+		elapsedTime := currentTime.Sub(startTime).Seconds()
+		if elapsedTime > 10 {
+			t.Fatalf("Timeout: Model version still has PENDING_REGISTERATION status after 10 seconds")
+		}
+		time.Sleep(2 * time.Second)
+	}
```

Comment:
You might be able to use something like this https://stackoverflow.com/a/52218264 here, instead of manually keeping track of time.

Reply:
discussed offline - tldr - not able to use testing methods through goroutine -- might require passing messages to channel and read them from the testMethod and call test fail (after checking the channel to see if message is returned) but maybe not worth it.